src/idiff/idiff.cpp

```diff
@@ -119,8 +119,9 @@ getargs(int argc, char* argv[])
 
 
 static bool
-read_input(const std::string& filename, ImageBuf& img, ImageCache* cache,
-           int subimage = 0, int miplevel = 0)
+read_input(const std::string& filename, ImageBuf& img,
+           std::shared_ptr<ImageCache> cache, int subimage = 0,
```

Comment:
On reading the reset of the review more carefully, it looks like ImageBuf, does need to keep a reference to the cache, so it looks like the by-value semantics are right here. Still, its worth keeping in mind the hidden cost of the reference count being twiddled at each function call (more than it would have been before).

Reply:
Specifically to this place in the code: it won't make a difference, this is in idiff, there is no concurrency here and only a few calls to this function per run.

But I take your point generally. In a case like this, do you think it's better to pass as `const shared_ptr<T>&` or just as a `T*`, knowing that the shared_ptr from which it came is guaranteed to outlive the function call?

Reply:
I will look, too, but please help me double check for instances of API calls that

1. Might be called very frequently.
2. Never need to take ownership or make a second, lasting, copy of the shared_ptr.

We should ensure that in such instances, we are passing a reference or a non-owning raw pointer.